src/Conversion/ONNXToTOSA/DialectBuilder.cpp

```diff
@@ -16,12 +16,91 @@
 #include "mlir/Dialect/Arith/IR/Arith.h"
 
 #include "src/Conversion/ONNXToTOSA/DialectBuilder.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
 #include "src/Dialect/ONNX/ONNXOps.hpp"
 
 using namespace mlir;
 
 namespace onnx_mlir {
 
+template <typename T>
+bool testNumberOfElementsMatch(ArrayRef<T> vec, ArrayRef<int64_t> shape) {
+  uint64_t numTotalElements = 1;
+  for (int64_t a : shape) {
+    numTotalElements *= a;
+  }
+  return (vec.size() == numTotalElements);
+}
+
+template <typename T>
+Value TosaBuilder::createConstFromRankedTensorAndVec(
+    ArrayRef<T> vec, RankedTensorType &constType) {
+  auto constAttr = DenseElementsAttr::get(constType, vec);
+
+  Value constOp =
+      rewriter().create<mlir::tosa::ConstOp>(loc(), constType, constAttr);
+  return constOp;
+}
+
+Value TosaBuilder::getConst(ArrayRef<int64_t> vec, ArrayRef<int64_t> shape) {
+
+  assert(testNumberOfElementsMatch(vec, shape) &&
+         "getConstTensor(): number of elements mismatch.");
+
+  auto constType = RankedTensorType::get(
+      shape, rewriter().getIntegerType(sizeof(int64_t) * 8));
+
+  Value constOp = this->createConstFromRankedTensorAndVec(vec, constType);
+  return constOp;
+}
+
+Value TosaBuilder::getConst(ArrayRef<int32_t> vec, ArrayRef<int64_t> shape) {
+
+  assert(testNumberOfElementsMatch(vec, shape) &&
+         "getConstTensor(): number of elements mismatch.");
+
+  auto constType = RankedTensorType::get(
+      shape, rewriter().getIntegerType(sizeof(int32_t) * 8));
+
+  Value constOp = this->createConstFromRankedTensorAndVec(vec, constType);
+  return constOp;
+}
+
+Value TosaBuilder::getConst(ArrayRef<float> vec, ArrayRef<int64_t> shape) {
+  assert(testNumberOfElementsMatch(vec, shape) &&
+         "getConstTensor(): number of elements mismatch.");
+
+  auto constType = RankedTensorType::get(shape, rewriter().getF32Type());
+
+  Value constOp = this->createConstFromRankedTensorAndVec(vec, constType);
+  return constOp;
+}
+
```

Comment:
There is a lot of common code here - did you consider having all three call a (template) function that is additionally passed the appropriate `rewriter().getF32Type()`/`rewriter().getIntegerType(sizeof(int32_t) * 8)`/...?

Reply:
Very nice idea! I will implement this